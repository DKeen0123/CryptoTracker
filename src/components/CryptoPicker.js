import React, { Component } from 'react';

class CryptoPicker extends Component {
  constructor(props) {
    super()
  }
  populateCryptos() {
    const { cryptos } = this.props;
    if (cryptos === null || cryptos === undefined) return '';
    if (Object.keys(cryptos).length === 0) return '';
    return cryptos.map(crypto => {
      return (
        <option id={crypto.id} key={crypto.id}>
          {crypto.name}
        </option>
      );
    });
  }

  render() {
    return (
      <div>
        <h3>Choose your crypto: </h3>
        <select
          onChange={this.props.selectCrypto}
          className="drop-down"
          value={this.props.chosenCrypto}
        >
          {this.populateCryptos()}
        </select>

      </div>
    );
  }
}

export default CryptoPicker;