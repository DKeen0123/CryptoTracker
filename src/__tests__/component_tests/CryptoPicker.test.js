import React from 'react';
import { shallow } from 'enzyme';
import CryptoPicker from '../../components/CryptoPicker';

describe('CryptoPicker', () => {
  const cryptoPicker = shallow(<CryptoPicker />);
  it('renders correctly', () => {
    expect(cryptoPicker).toMatchSnapshot();
  });
});
