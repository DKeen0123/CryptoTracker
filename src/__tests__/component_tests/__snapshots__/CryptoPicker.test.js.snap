// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoPicker renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CryptoPicker />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Choose your crypto: 
</h3>,
        <select
          className="drop-down"
          onChange={undefined}
          value={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Choose your crypto: ",
        },
        "ref": null,
        "rendered": "Choose your crypto: ",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "drop-down",
          "onChange": undefined,
          "value": undefined,
        },
        "ref": null,
        "rendered": "",
        "type": "select",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Choose your crypto: 
</h3>,
          <select
            className="drop-down"
            onChange={undefined}
            value={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Choose your crypto: ",
          },
          "ref": null,
          "rendered": "Choose your crypto: ",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "drop-down",
            "onChange": undefined,
            "value": undefined,
          },
          "ref": null,
          "rendered": "",
          "type": "select",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
